tikzplotlib/_text.py

```diff
@@ -140,58 +139,91 @@ def _transform_positioning(ha, va):
-            "<-": ["<-"],
-            "<->": ["<->"],
-            "|-|": ["|-|"],
-            "-|>": ["-latex"],
-            "<|-": ["latex-"],
-            "<|-|>": ["latex-latex"],
-            "]-[": ["|-|"],
-            "-[": ["-|"],
-            "]-": ["|-"],
-            "fancy": ["-latex", "very thick"],
-            "simple": ["-latex", "very thick"],
-            "wedge": ["-latex", "very thick"],
-        }
-        arrow_style = []
-        if obj.arrowprops is not None:
-            if obj.arrowprops["arrowstyle"] is not None:
-                if obj.arrowprops["arrowstyle"] in arrow_translate:
-                    arrow_style += arrow_translate[obj.arrowprops["arrowstyle"]]
-                    data, col, _ = _color.mpl_color2xcolor(
-                        data, obj.arrow_patch.get_ec()
-                    )
-                    arrow_style.append(col)
-
-        ff = data["float format"]
-        arrow_fmt = (
-            "\\draw[{}] (axis cs:"
-            + ff
-            + ","
-            + ff
-            + ") -- (axis cs:"
-            + ff
-            + ","
-            + ff
-            + ");\n"
-        )
-        the_arrow = arrow_fmt.format(
-            ",".join(arrow_style), ann_xytext[0], ann_xytext[1], ann_xy[0], ann_xy[1]
-        )
+        return obj.get_position()
+
+    # special cases only for text_coords
+    if ann_textcoords == "offset points":
+        text_pos = "{{}} ++({ff}pt,{ff}pt)".format(ff=ff).format(xy_pos, *ann_xytext)
+    elif ann_textcoords == "offset pixels":
+        text_pos = "{{}} ++({ff}px,{ff}px)".format(ff=ff).format(xy_pos, *ann_xytext)
+    else:
+        try:
+            text_pos = _parse_annotation_coords(ff, ann_xycoords, ann_xytext)
+        except NotImplementedError:
+            # Anything else except for explicit positioning is not supported yet
+            return obj.get_position()
+
+    # Create a basic tikz arrow
+    arrow_translate = {
+        "-": ["-"],
+        "->": ["->"],
+        "<-": ["<-"],
+        "<->": ["<->"],
+        "|-|": ["|-|"],
+        "-|>": ["-latex"],
+        "<|-": ["latex-"],
+        "<|-|>": ["latex-latex"],
+        "]-[": ["|-|"],
+        "-[": ["-|"],
+        "]-": ["|-"],
+        "fancy": ["-latex", "very thick"],
+        "simple": ["-latex", "very thick"],
+        "wedge": ["-latex", "very thick"],
+    }
+    arrow_style = []
+    if obj.arrowprops is not None:
+        if obj.arrowprops["arrowstyle"] is not None:
+            if obj.arrowprops["arrowstyle"] in arrow_translate:
+                arrow_style += arrow_translate[obj.arrowprops["arrowstyle"]]
+                data, col, _ = _color.mpl_color2xcolor(
+                    data, obj.arrow_patch.get_ec()
+                )
+                arrow_style.append(col)
+
+    if arrow_style:
+        the_arrow = (
+            "\\draw[{}] {} -- {};\n"
+        ).format(",".join(arrow_style), text_pos, xy_pos)
         content.append(the_arrow)
```

Comment:
Now only draws arrows if one was requested - annotations by default do not have any line at all.